main.cpp

```diff
@@ -0,0 +1,190 @@
+    std::string emailAddress = "";
+    std::string password = "";
+    std::string timezoneOffset = "";
+    std::string knownrecipientsFilePath = "";
+
+    // Process information based on user entry (init or edit)
+
+    // Initialize mode
+    if (static_cast<std::string>(argv[1]) == "init") {
+
+        // Get first name
+        do {
+            std::cout << "Please enter your first name: "<<std::endl;
+            std::getline(std::cin, firstName);
+        } while (firstName == "");
+
+        // Get last name
+        do {
+            std::cout << "Please enter your last name: "<<std::endl;
+            std::getline(std::cin,lastName);
+        } while (lastName == "");
+
+        // Get email address
+        do {
+            std::cout << "Please enter your e-mail address: "<<std::endl;
+            std::getline(std::cin,emailAddress);
+        } while (emailAddress == "");
+
+        // Get password
+        do {
+            std::cout << "Please enter your password: "<<std::endl;
+            std::getline(std::cin,password);
+        } while (password == "");
+
+        // Get time zone offset
+        do {
+            std::cout << "Please enter your time zone offset: "<<std::endl;
+            std::getline(std::cin,timezoneOffset);
+        } while (timezoneOffset == "");
+
+        // Get knownrecipients file path
+        std::cout <<"Please enter file path for knownrecipients file (If no file path is entered, a default file path "
+                "will be set): "<<std::endl;
+            std::getline(std::cin,knownrecipientsFilePath);
+            if (knownrecipientsFilePath == "") {
+                knownrecipientsFilePath = DEFAULT_FILE_PATH;
+            }
+
+        // Open and Write initialize data to file
+        std::ofstream outputFile;
+
+        outputFile.open(knownrecipientsFilePath);
+
+        std::cout << "Writing initialize data to file.\n";
+        outputFile << firstName << std::endl;
+        outputFile << lastName << std::endl;
+        outputFile << emailAddress << std::endl;
+        outputFile << password << std::endl;
+        outputFile << timezoneOffset << std::endl;
+        outputFile << knownrecipientsFilePath << std::endl;
+
+        // Close the open file
+        outputFile.close();
+    }
+
+    else if (static_cast<std::string>(argv[1]) == "edit") {
+        // Edit mode
+        std::string editField = static_cast<std::string>(argv[2]);
+        int editFieldNum = 0;
+
+        if (editField == "firstName") {
+            editFieldNum = 1;
+        }
+        if (editField == "lastName") {
+            editFieldNum = 2;
+        }
+        if (editField == "emailAddress") {
+            editFieldNum = 3;
+        }
+        if (editField == "password") {
+            editFieldNum = 4;
+        }
+        if (editField == "timezoneOffset") {
+            editFieldNum = 5;
+        }
+        if (editField == "knownrecipientsFilePath") {
+        editFieldNum = 6;
```

Comment:
For exploration of switch verses if else in this section I can understand, but you have duplicated the selection logic by transforming strings to integers and then back to the logic. You could just as easily use only the if else.